src/regular_cartesian_mesh_filter.cpp

```diff
@@ -0,0 +1,501 @@
+
+    case -3:
+      k--;
+      on[2] = 1;
+      break;
+
+    case 3:
+      k++;
+      on[2] = -1;
+      break;
+
+    default:
+      break;
+  }
+}
+
+std::pair<double, int> RegularCartesianMeshFilter::distance_to_next_index(
+    const Position& r, const Direction& u, const std::array<int, 3>& on, int i,
+    int j, int k) {
+  // Get position at center of current tile
+  double xc = r_low_.x() + i * dx_ + 0.5 * dx_;
+  double yc = r_low_.y() + j * dy_ + 0.5 * dy_;
+  double zc = r_low_.z() + k * dz_ + 0.5 * dz_;
+
+  // Get relative position in cell
+  Position r_tile(r.x() - xc, r.y() - yc, r.z() - zc);
+
+  // Set our initial value for the distance and the index change
+  double dist = INF;
+  int key = 0;
+
+  // Check all six sides
+  const double diff_xl = -dx_ * 0.5 - r_tile.x();
+  const double diff_xh = dx_ * 0.5 - r_tile.x();
+  const double diff_yl = -dy_ * 0.5 - r_tile.y();
+  const double diff_yh = dy_ * 0.5 - r_tile.y();
+  const double diff_zl = -dz_ * 0.5 - r_tile.z();
+  const double diff_zh = dz_ * 0.5 - r_tile.z();
+
+  const double ux_inv = 1. / u.x();
+  const double uy_inv = 1. / u.y();
+  const double uz_inv = 1. / u.z();
+
+  const double d_xl = diff_xl * ux_inv;
+  const double d_xh = diff_xh * ux_inv;
+  const double d_yl = diff_yl * uy_inv;
+  const double d_yh = diff_yh * uy_inv;
+  const double d_zl = diff_zl * uz_inv;
+  const double d_zh = diff_zh * uz_inv;
+
+  if (d_xl > 0. && d_xl < dist && on[0] != -1) {
+    dist = d_xl;
+    key = -1;
+  }
+
+  if (d_xh > 0. && d_xh < dist && on[0] != 1) {
+    dist = d_xh;
+    key = 1;
+  }
+
+  if (d_yl > 0. && d_yl < dist && on[1] != -1) {
+    dist = d_yl;
+    key = -2;
+  }
+
+  if (d_yh > 0. && d_yh < dist && on[1] != 1) {
+    dist = d_yh;
+    key = 2;
+  }
+
+  if (d_zl > 0. && d_zl < dist && on[2] != -1) {
+    dist = d_zl;
+    key = -3;
+  }
+
+  if (d_zh > 0. && d_zh < dist && on[2] != 1) {
+    dist = d_zh;
+    key = 3;
+  }
+
+  return {dist, key};
+}
+
+// Make the cartesian or position filter class
+std::shared_ptr<RegularCartesianMeshFilter> make_mesh_position_filter(
+    const YAML::Node& node) {
+  if (!node["low"])
+    fatal_error(
+        "For mesh position-filter \"low\" co-ordinates is not provided.");
+  if (!node["high"])
+    fatal_error(
+        "For mesh position-filter \"high\" co-ordinates is not provided.");
+  if (!node["shape"])
+    fatal_error("For mesh position-filter \"shape\" is not provided.");
+
+  std::vector<double> low_point = node["low"].as<std::vector<double>>();
+  std::vector<double> high_point = node["high"].as<std::vector<double>>();
+  std::vector<std::size_t> shape_ =
+      node["shape"].as<std::vector<std::size_t>>();
+
+  if (shape_[0] == 1 && shape_[1] == 1 && shape_[2] == 1) {
+    std::string name_ = node["name"].as<std::string>();
+    fatal_error("For " + name_ +
+                " tally, the shape [1,1,1] with reactilinear mesh position "
+                "filter is used instead of box filter.");
```

Comment:
```suggestion
    fatal_error("For " + name_ +
                " tally, the shape [1,1,1] with RegularCartesianMesh "
                "filter is used instead of box filter.");
```